packages/datepicker/src/react/calendar.tsx

```diff
@@ -64,11 +69,22 @@ interface CalendarProps
   extends React.HTMLAttributes<HTMLDivElement>,
     Pick<RenderProps, 'calendars' | 'getBackProps' | 'getForwardProps'> {
   slide?: ValueOf<typeof slides>
+  uniqueId?: (prefix: string) => string
+  selected?: Date
 }
 
 export const Calendar = React.forwardRef<HTMLDivElement, CalendarProps>(
   (
-    { calendars, getBackProps, getForwardProps, children, slide, ...rest },
+    {
+      calendars,
+      getBackProps,
+      getForwardProps,
+      children,
+      slide,
+      uniqueId = defaultUniqueId,
```

Comment:
This is the current approach to solving this problem for snapshots. It's a bit painful, wouldn't you agree?

I would eventually love another solution. If you have ideas you like, I'd love to hear them.

(This is a musing, not an ask for change in this PR.)

Reply:
Not sure honestly my normal approach for this is to overwrite a global but I can't see a reasonable way around this other than mimicking downshifts approach which is an iterator I believe.

Reply:
Yeah, I like the iterator better. Outside scope here, if wanted, but I would love to move to it.